Tests/IntegrationTests.Shared/ObjectIntegrationTests.cs

```diff
@@ -247,6 +247,23 @@ public void IteratePeople()
             Assert.That (iterCount, Is.EqualTo (3));
         }
 
+        [Test]
+        public void IsValidReturnsTrueWhenObjectRowIsAttached()
+        {
+            _realm.Write(() =>
+            {
+                var p1 = _realm.CreateObject<Person>();
+                Assert.That(p1.IsValid);
+
+                _realm.Remove(p1);
+                Assert.That(p1.IsValid, Is.False);
+            });
+
+            // IsValid should always return false for standalone objects
```

Comment:
Hmm, seems that Java got these semantics wrong then. We only return `true` for `isValid` if it is managed and not deleted. I could see the sense in changing it to return true for unmanaged objects as well since data is actually readable.